opentracing-spring-messaging/src/main/java/io/opentracing/contrib/spring/integration/messaging/OpenTracingChannelInterceptor.java

```diff
@@ -66,33 +67,44 @@ public OpenTracingChannelInterceptor(Tracer tracer) {
       spanBuilder.asChildOf(extractedContext);
     }
 
-    Span span = spanBuilder.startActive(true).span();
-
-    if (isConsumer) {
-      log.trace("Adding 'messageConsumed' header");
-      carrier.put(Headers.MESSAGE_CONSUMED, "true");
-      // TODO maybe we should remove Headers.MESSAGE_SENT_FROM_CLIENT header here?
-    } else {
-      log.trace("Adding 'messageSent' header");
-      carrier.put(Headers.MESSAGE_SENT_FROM_CLIENT, "true");
+    Span span = spanBuilder.start();
+    Scope scope = tracer.activateSpan(span);
+    carrier.addHeader(SCOPE_HEADER, scope);
+    try {
+      if (isConsumer) {
+        log.trace("Adding 'messageConsumed' header");
+        carrier.put(Headers.MESSAGE_CONSUMED, "true");
+        // TODO maybe we should remove Headers.MESSAGE_SENT_FROM_CLIENT header here?
+      } else {
+        log.trace("Adding 'messageSent' header");
+        carrier.put(Headers.MESSAGE_SENT_FROM_CLIENT, "true");
+      }
+      log.trace(String.format("Pre-send: starting a new span %s , carrier extracted context %s", span, extractedContext));
+
+      tracer.inject(span.context(), Format.Builtin.TEXT_MAP, carrier);
+      return carrier.getMessage();
+    } catch (Exception e) {
```

Comment:
Change this in a separate PR

Reply:
Ok, I have removed this part.

Reply:
It seems it's back here. Can you please keep only changes related to API update?

Reply:
@pavolloffay  Done. We still risk to have an open scope in case there is an exception in our code or in the spring stack (like they don't invoke us in case of exception), but I guess we should be able to rely on Spring and us.